snapcraft/internal/sources.py

```diff
@@ -474,6 +475,50 @@ def provision(self, dst, clean_target=True, keep_deb=False):
             os.remove(deb_file)
 
 
+class Rpm(FileBase):
+
+    def __init__(self, source, source_dir, source_tag=None, source_commit=None,
+                 source_branch=None, source_depth=None):
+        super().__init__(source, source_dir, source_tag, source_commit,
+                         source_branch, source_depth)
+        if source_tag:
+            raise IncompatibleOptionsError(
+                'can\'t specify a source-tag for a rpm source')
+        elif source_commit:
+            raise IncompatibleOptionsError(
+                'can\'t specify a source-commit for a rpm source')
+        elif source_branch:
+            raise IncompatibleOptionsError(
+                'can\'t specify a source-branch for a rpm source')
+
+    def provision(self, dst, clean_target=True, keep_rpm=False):
+        rpm_file = os.path.join(self.source_dir, os.path.basename(self.source))
+
+        if clean_target:
+            tmp_rpm = tempfile.NamedTemporaryFile().name
+            shutil.move(rpm_file, tmp_rpm)
+            shutil.rmtree(dst)
+            os.makedirs(dst)
+            shutil.move(tmp_rpm, rpm_file)
+
+        # Ensure dst does not have trailing slash
+        dst = dst.rstrip('/')
+        # Open the RPM file and extract it to destination
+        with libarchive.file_reader(rpm_file) as rpm:
+            for rpm_file_entry in rpm:
+                # Binary RPM archive data has paths starting with ./ to support
+                # relocation if enabled in the building of RPMs
+                rpm_file_entrypath = rpm_file_entry.pathname.lstrip('./')
+                rpm_file_entrypath = rpm_file_entrypath.lstrip('/')
```

Comment:
Not crazy until proven otherwise :)
Thanks for adding the tests. I'm wondering about these two lines.
Is there a way to make libarchive.file_writer to include entries with ./ and / ?